Magento2/Sniffs/Security/InsecureFunctionSniff.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Functions;
+
+use PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff;
+
+/**
+ * Detects the use of insecure functions.
+ */
+class InsecureFunctionSniff extends ForbiddenFunctionsSniff
+{
+    /**
+     * If true, an error will be thrown; otherwise a warning.
+     *
+     * @var boolean
+     */
+    public $error = false;
```

Comment:
Why is it a warning?

Reply:
It's under `Severity 9 warnings: Possible security and issues that may cause bugs.` section.
We cannot reject extensions because of such functions right now. We will work iteratively to make rules stricter.